tests/baselines/reference/cyclicGenericTypeInstantiationInference.types

```diff
@@ -2,8 +2,8 @@
 
 === cyclicGenericTypeInstantiationInference.ts ===
 function foo<T>() {
->foo : <T>() => { y2: { y2: { y2: { y2: { y2: { y2: { y2: { y2: { y2: { y2: { y2: any; }; }; }; }; }; }; }; }; }; }; }
->    : ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+>foo : <T>() => { y2: typeof z; }

```

Comment:
At first you'd think this is good; but `z` isn't actually in scope...? And isn't a parameter?